### ISSUE TITLE
React component should be inlined, but isn't being inlined

### ISSUE THREAD

**Issue by trueadm** [facebook/prepack#2497]
Given the case below, both child components should be evaluated and inlined:

```js
var React = require("react");

function Child() {
  return <div>This should be inlined</div>;
}

function Child2() {
  return <span>This should be inlined too</span>;
}

function App(props) {
  var a = props.x ? null : <Child2 />;
  return a && <Child />;
}

App.getTrials = function(renderer, Root) {
  renderer.update(<Root />);
  return [["simple conditions", renderer.toJSON()]];
};

if (this.__optimizeReactComponentTree) {
  __optimizeReactComponentTree(App);
}

module.exports = App;
```

**Comment by trueadm** [comment]
This issue is actually fixed with https://github.com/facebook/prepack/pull/2494

**Comment by trueadm** [comment]
Fixed in #2494
